### PR TITLE
Add JSON formatter to Storage Queue tracing

### PR DESCRIPTION
**Pull request by tniek**
- This PR adds a JSON formatter to the Storage Queue and makes application tracing (and eventually, ingestion of customer metrics) much more flexible.
- The `queueStorageLogHandler` now uses by default the JsonFormatter which outputs the entire trace message as a JSON-formatted string; custom fields (like salmonid or payloadVersion) are supported as well.
- Sample KQL query:
```
SapMonitorActivity
| project message
| extend p=parse_json(message)
| extend pid=p.pid
| extend timestamp=p.timestamp
| extend traceLevel=p.traceLevel
| extend msg=p.msg
| extend module=p.module
| extend lineNum=p.lineNum
| extend function=p.function
| extend sapmonId=p.sapmonId
| extend payloadVersion=p.payloadVersion
| where sapmonId == "bbbbbbbaaaaaaa"
| order by tostring(timestamp) asc
| project-away p, message, sapmonId
| project-reorder pid, timestamp, traceLevel, msg, module, lineNum, function, payloadVersion
```

### CODE REVIEW

File: sapmon/payload/helper/tracing.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 # Python modules
 import argparse
+from collections import OrderedDict
```
Comment (persiaAziz) [inline sapmon/payload/helper/tracing.py]: Do we need to pip install this package? If so, please add it to the configure VM shell script

File: sapmon/payload/helper/tracing.py [hunk 2]
```diff
@@ -12,12 +14,65 @@
 from const import *
 from helper.azure import *
 
+# Formats a log/trace payload as JSON-formatted string
+class JsonFormatter(logging.Formatter):
+   def __init__(self, fields = [], datefmt=None, customJson=None):
```
Comment (persiaAziz) [inline sapmon/payload/helper/tracing.py]: Please add comments specifying the type of the parameters passed in this constructor

File: sapmon/payload/helper/tracing.py [hunk 3]
```diff
@@ -12,12 +14,65 @@
...
+   def _formatTime(self,
+                   record: logging.LogRecord) -> None:
+      if self.usesTime():
+         record.asctime = self.formatTime(record, self.datefmt)
+
+   # Combines any supplied fields with the log record msg field into an object to convert to JSON 
+   def _getJsonData(self,
+                    record: logging.LogRecord) -> OrderedDict():
+      if len(self.fields.keys()) > 0:
+         fields = []
```
Comment (persiaAziz) [inline sapmon/payload/helper/tracing.py]: Please pick a different name for this "fields" to increase readability.

File: sapmon/payload/helper/tracing.py (hunk 1 above)
Comment (tniek) [reply sapmon/payload/helper/tracing.py]: Nope, this is a standard package that ships with any Python version >= 2.4

File: sapmon/payload/helper/tracing.py (hunk 3 above)
Comment (tniek) [reply sapmon/payload/helper/tracing.py]: Done

File: sapmon/payload/helper/tracing.py (hunk 2 above)
Comment (tniek) [reply sapmon/payload/helper/tracing.py]: Done

File: sapmon/payload/helper/tracing.py (hunk 3 above)
Comment (persiaAziz) [reply sapmon/payload/helper/tracing.py]: I am still seeing the variable name to be fields? Did you forget to push?

**Pull request merged**